File: 78-subsets/subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> subsets = new ArrayList<>();
        subsets.add(new ArrayList<>());

        for (int val : nums) {
            int size = subsets.size();
            for (int i = 0; i < size; i++) {
                List<Integer> subset = new ArrayList<>(subsets.get(i));
                subset.add(val);
                subsets.add(subset);
            }
        }

        return subsets;
    }
}

File: 78-subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        
        def function(idx, path):
            if idx == len(nums):
                res.append(path[:])
                return
            
            path.append(nums[idx])
            function(idx+1, path)
            path.pop()

            function(idx + 1, path)

        function(0, [])

        return res    
            

